src/app/games/quarto/QuartoRules.ts

```diff
@@ -129,41 +129,175 @@ interface LineInfos {
     boardStatus: MGPOptional<BoardStatus>;
 }
 
-export class QuartoRules extends Rules<QuartoMove, QuartoState> {
+abstract class VictoryPattern {
+
+    protected constructor(public readonly level: number,
+                          public readonly coordPattern: CoordSet,
+                          public readonly initialCoord: Coord,
+                          public readonly remaining: number,
+    ) {}
+
+    public abstract getNextPattern(): VictoryPattern;
```

Comment:
wtf is this? pas fan du pattern pantoute, si tu veux créer 4 lignes verticales, fait une boucle for sur x plutôt que de faire une boucle for sur i et faire un getNextPattern 4 fois